Xamarin.Forms.Platform.iOS/CollectionView/CarouselViewDataSource.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using Foundation;
+using UIKit;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	internal class CarouselViewDataSource : UICollectionViewDataSource
+	{
+		readonly ItemsView _itemsView;
+		readonly ItemsViewLayout _itemsViewLayout;
+		readonly IItemsViewSource _source;
+		readonly string _reuseId;
+		int _indexOffset = 0;
+
+		public CarouselViewDataSource(ItemsView itemsView, ItemsViewLayout itemsViewLayout, IItemsViewSource source, string reuseId)
+		{
+			_itemsView = itemsView;
+			_itemsViewLayout = itemsViewLayout;
+			_source = source;
+			_reuseId = reuseId;
+		}
+
+		public UICollectionViewCell CellForItemAtIndexPath(UICollectionView collectionView, NSIndexPath dequeueIndexPath, NSIndexPath usableIndexPath)
+		{
+			var cell = collectionView.DequeueReusableCell(_reuseId, dequeueIndexPath) as UICollectionViewCell;
+			UpdateTemplatedCell(cell as TemplatedCell, usableIndexPath);
+			return cell;
+		}
+
+		public void ShiftContentArray(int shiftCells)
+		{
+			var correctedIndex = GetCorrectedIndex(shiftCells);
+			_indexOffset += correctedIndex;
+		}
+
+		protected virtual void UpdateTemplatedCell(TemplatedCell cell, NSIndexPath indexPath)
+		{
+			//cell.ContentSizeChanged -= CellContentSizeChanged;
```

Comment:
Was commenting this out intentional?

Reply:
need to find a way to reuse the existing code, right now i just copy pasted it 
that said the CotnentSize on a Carousel would be updated on a cell right?